Source/RapyutaSimulationPlugins/Private/Drives/RobotVehicleMovementComponent.cpp

```diff
@@ -149,6 +347,72 @@ void URobotVehicleMovementComponent::TickComponent(float InDeltaTime,
-                      FVector(OdomData.pose_pose_position_x, OdomData.pose_pose_position_y, OdomData.pose_pose_position_z));
+    FTransform TF;
+    FVector Pos(OdomData.pose_pose_position_x, OdomData.pose_pose_position_y, OdomData.pose_pose_position_z);
+    TF.SetTranslation(Pos);
+    TF.SetRotation(OdomData.pose_pose_orientation);
+
+    return TF;
+}
+
+void URobotVehicleMovementComponent::InitMovementComponent()
+{
+    InitOdom();
+
+    TArray<UActorComponent*> ActorContactPoints = PawnOwner->GetComponentsByTag(USceneComponent::StaticClass(), "ContactPoint");
+    for (auto ACP : ActorContactPoints)
+    {
+        USceneComponent* SCP = Cast<USceneComponent>(ACP);
+        if (SCP)
+            ContactPoints.Add(SCP);
+    }
+    UE_LOG(LogTemp,
+           Warning,
+           TEXT("URobotVehicleMovementComponent::InitMovementComponent - Nb Contact Points : %d"),
+           ContactPoints.Num());
+
+    // Compute the starting distance between the robot root and the floor
+    // We consider that the robot is on a horizontal floor at the beginning
+    FCollisionQueryParams TraceParams = FCollisionQueryParams(FName(TEXT("Contact_Trace")), true, PawnOwner);
+    TraceParams.bReturnPhysicalMaterial = true;
+    TraceParams.bTraceComplex = true;
+    TraceParams.bReturnFaceIndex = true;
+    TraceParams.AddIgnoredActor(PawnOwner);
+
+    FVector startPos = PawnOwner->GetActorLocation() + FVector(0., 0., 10.);
+    FVector endPos = PawnOwner->GetActorLocation() - FVector(0., 0., 50.);
+    FHitResult HitResult;
+    bool IsFloorHit = GetWorld()->LineTraceSingleByChannel(HitResult,
+                                                           startPos,
+                                                           endPos,
+                                                           ECollisionChannel::ECC_Visibility,
+                                                           TraceParams,
+                                                           FCollisionResponseParams::DefaultResponseParam);
+    if (IsFloorHit)
+    {
+        MinDistanceToFloor = HitResult.Distance;
+        UE_LOG(LogTemp,
+               Warning,
+               TEXT("URobotVehicleMovementComponent::InitMovementComponent - Min Distance To Floor = %f"),
+               MinDistanceToFloor);
+    }
+}
+
+void URobotVehicleMovementComponent::SetMovingPlatform(AActor* platform)
+{
+    // UE_LOG(LogTemp, Warning, TEXT("URobotVehicleMovementComponent::SetMovingPlatform..."));
+    MovingPlatform = platform;
+    LastPlatformLocation = FVector(platform->GetActorLocation());
+    LastPlatformRotation = FQuat(platform->GetActorQuat());
+}
+
+bool URobotVehicleMovementComponent::IsOnMovingPlatform()
+{
+    return MovingPlatform != nullptr;
+}
+
+void URobotVehicleMovementComponent::RemoveMovingPlatform()
+{
+    // UE_LOG(LogTemp, Warning, TEXT("URobotVehicleMovementComponent::RemoveMovingPlatform..."));
+    MovingPlatform = nullptr;

```

Comment:
Probably this means `Detach from MovingPlatform`, or otherwise it should be `MovingPlatform->Destroy()`?